Makefile

```diff
@@ -0,0 +1,53 @@
+.PHONY: all index qnagen query eval help
+.DEFAULT_GOAL := help
+
+# Load .env file if exists and export all variables before running any target
+ENV_FILE := .env
+ifeq ($(filter $(MAKECMDGOALS),config clean),)
+	ifneq ($(strip $(wildcard $(ENV_FILE))),)
+		ifneq ($(MAKECMDGOALS),config)
+			include $(ENV_FILE)
+			export
+		endif
+	endif
+endif
```

Comment:
do we still need load_env?

Reply:
You mean the `load_env` calls within the Python code?

Yes...you need both. In fact, the calls within the Python are probably more important than this `Makefile` call.

This `Makefile` load of .env files should be there to ensure that bash scripts and other command line calls made within this `Makefile` have their env variables loaded. It's there to prevent someone having to `source` the `.env` directly. The `Makefile` shouldn't really know how the python code is written. So, this script helps the `Makefile` run, but *if the `Makefile` doesn't need or use any environment variables I'd argue that this part of the script isn't needed.

The Python files then load their own .env requirements. We can't be use that the Python code will be called from the `Makefile`, so the use of `dotenv` within the code remains relevant. This is especially true for things like tests that won't invoke python code via the `Makefile`

So, I see this addition to the `Makefile` as a useful safety net for Python code, and also a way to reference env variables from within the `Makefile` or any bash scripts it calls. I don't see it as a replacement for the `load_dotenv` calls.

Reply:
Thanks, @martinpeck.
closing this as we all agree.

Reply:
Sorry I wasn't clear, I was referring to this load_env command in the Makfile:
```
load_env: ## 📃 Load .env file
	$(call target_title, "loading env file") \
	&& source .env
```